diff --git a/CMakeLists.txt b/CMakeLists.txt
index 38dcb60..ef9c2bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,26 @@
+## CMake configuration file of glog project
+##
+## When this project is a subproject. The "glog::glog" target is in this case
+## an ALIAS library target for the "glog" library target. Targets which depend
+## on the glog library should link to the "glog::glog" library target.
+##
+## Example CMakeLists.txt of user project which requires separate glog installation:
+##   cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+##   project(Foo)
+##
+##   find_package(glog REQUIRED)
+##
+##   add_executable(foo src/foo.cc)
+##   target_link_libraries(foo glog::glog)
+##
+## Example CMakeLists.txt of super-project which contains glog source tree:
+##   cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+##   project(Foo)
+##
+##   add_subdirectory(glog)
+##
+##   add_executable(foo src/foo.cc)
+##   target_link_libraries(foo glog::glog)
 cmake_minimum_required (VERSION 3.0)
 
 if (POLICY CMP0042)
@@ -483,7 +506,7 @@ if (WIN32 AND HAVE_SNPRINTF)
 endif (WIN32 AND HAVE_SNPRINTF)
 
 if (gflags_FOUND)
-  target_link_libraries (glog PUBLIC gflags)
+  target_link_libraries (glog PUBLIC gflags::gflags)
 
   if (NOT BUILD_SHARED_LIBS)
     # Don't use __declspec(dllexport|dllimport) if this is a static build
